File: 1154-day-of-the-year/python3/main.py
class Solution:
    def dayOfYear(self, date: str) -> int:
        year, month,day = map(int, date.split('-'))
        monthDays = [31,28,31,30,31,30,31,31,30,31,30,31]
        if year % 4 == 0 and year % 100 != 0 or year % 400 == 0:
            monthDays[1] = 29
        return sum(monthDays[:month-1]) + day

File: 1154-day-of-the-year/go/main.go
func dayOfYear(date string) int {
	year, _ := strconv.Atoi(date[:4])
	month, _ := strconv.Atoi(date[5:7])
	day, _ := strconv.Atoi(date[8:])
	monthDays := []int{31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31}

	for i := 0; i < month-1; i++ {
		day += monthDays[i]
	}
	if month > 2 && (year%400 == 0 || (year%4 == 0 && year%100 != 0)) {
		day++
	}
	return day
}
